pkg/cmd/image/push.go

```diff
@@ -149,10 +150,21 @@ func Push(ctx context.Context, client *containerd.Client, rawRef string, options
 		return err
 	}
 
-	if err = signutil.Sign(rawRef, options.GOptions.Experimental, options.SignOptions); err != nil {
+	if img, err := client.ImageService().Get(ctx, ref); err != nil {
 		return err
+	} else {
+		refSpec, err := reference.Parse(ref)
+		if err != nil {
+			return err
+		}
+		signRef := fmt.Sprintf("%s@%s", refSpec.String(), img.Target.Digest.String())
```

Comment:
ref may already contain the digest?

Reply:
how? AFAIK, the digest will be created after the image is pushed, so people can't give a digest to the reference.

Reply:
The local images have digests too

Reply:
I tried to use ref that contains digest and got the following error while pushing it:
```shell
$ COSIGN_PASSWORD="" _output/nerdctl push --sign cosign --cosign-key cosign.key developerguy/melange:latest@sha256:be956ef9dfe313806ec655e7eaffbc9333522c3560acc56a2dc833575c51cd6a
FATA[0000] failed to create a tmp single-platform image "docker.io/developerguy/melange@sha256:be956ef9dfe313806ec655e7eaffbc9333522c3560acc56a2dc833575c51cd6a-tmp-reduced-platform": image "
docker.io/developerguy/melange@sha256:be956ef9dfe313806ec655e7eaffbc9333522c3560acc56a2dc833575c51cd6a": not found
```

but I do have that image:

```shell
lima@lima-nerdctl:/Users/batuhan.apaydin/workspace/projects/personal/poc/nerdctl$ _output/nerdctl images --digests
REPOSITORY              TAG       DIGEST                                                                     IMAGE ID        CREATED         PLATFORM       SIZE        BLOB SIZE
developerguy/melange    latest    sha256:be956ef9dfe313806ec655e7eaffbc9333522c3560acc56a2dc833575c51cd6a    be956ef9dfe3    16 hours ago    linux/amd64    52.6 MiB    23.3 MiB
```

Reply:
This is a bug, and it has to be eventually supported

Reply:
so, is this a blocker thing against merging this PR?

Reply:
No